### PR TITLE
Move tag search from other branch and add tests

### PR DESCRIPTION
**Pull request by rebekadittrich**

### CODE REVIEW

File: src/main/java/com/google/sps/data/Search.java [hunk 1]
```diff
@@ -23,18 +26,20 @@
   private final String end;
   private final String duration;
   private final String location;
+  private final List<String> tags;
 
   public Search() {
-    this(null, null, null, null, null, null);
+    this(null, null, null, null, null, null, null);
```
Comment (LadyPython) [inline src/main/java/com/google/sps/data/Search.java:32]: What will be if you just leave in undefined?

**Review by LadyPython — APPROVED** [review]

**Review by andru47 — APPROVED** [review]

File: src/main/java/com/google/sps/data/EventStorage.java [hunk 2]
```diff
@@ -86,15 +95,35 @@ public Event getEvent(String eventId) {
...
+    for (Integer count : counts) {
+      for (Map.Entry<Event, Integer> eventWithCount : eventsWithCounts.entrySet()) {
+        if (eventWithCount.getValue().equals(count)) {
+          events.add(eventWithCount.getKey());
+        }
+      }
+    }
+
     return events;
   }
```
Comment (marcbury-g) [inline src/main/java/com/google/sps/data/EventStorage.java]: This looks a bit hacky. Maybe have a look here how to sort a map by value: https://dzone.com/articles/how-to-sort-a-map-by-value-in-java-8

File: src/main/java/com/google/sps/data/Search.java [hunk 3]
```diff
@@ -80,6 +100,16 @@ public boolean eventInCategory(String category) {
     || category.equals(this.category);
   }
 
+  public int countMatchingTags(List<String> tags) {
+    if (this.tags == null || this.tags.size() == 0) {
+      return -1;
```
Comment (marcbury-g) [inline src/main/java/com/google/sps/data/Search.java:105]: Why -1 and not 0?

File: src/main/java/com/google/sps/data/Search.java [hunk 4]
```diff
@@ -80,6 +100,16 @@ public boolean eventInCategory(String category) {
     || category.equals(this.category);
   }
 
+  public int countMatchingTags(List<String> tags) {
```
Comment (pmueller-gg) [inline src/main/java/com/google/sps/data/Search.java:103]: What is the reason to return -1 for `this.tags.size() == 0`?

**Review by pmueller-gg — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/sps/data/Search.java (hunk 3 above)
Comment (rebekadittrich) [reply src/main/java/com/google/sps/data/Search.java:105]: To distinguish between empty search and empty tags for an events. The event should be added for the list if the tag search is empty and shouldn't when the event's list of tags is empty.

File: src/main/java/com/google/sps/data/Search.java (hunk 4 above)
Comment (rebekadittrich) [reply src/main/java/com/google/sps/data/Search.java:103]: To distinguish between empty search and empty tags for an events. The event should be added for the list if the tag search is empty and shouldn't when the event's list of tags is empty.

**Review by marcbury-g — APPROVED** [review]

**Review by pmueller-gg — APPROVED** [review]

**Pull request merged**